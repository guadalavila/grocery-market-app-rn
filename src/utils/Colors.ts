export const Colors = {
    primary: '#68B984',
    primaryLight: '#CFFDE1',
    greenUltraLight: '#E9F9F2',
    green: '#2DA32A',
    secondary: '#3D5656',
    yellow: '#FED049',
    orange: '#FCB753',
    white: '#FFFFFF',
    black: '#000000',
    greyLight: '#C5C5C5',
    greyUltraLight: '#F7F8FA',
    background: '#FCFDFB',
    grey: '#9f9f9f',
    divider: '#EFEEF1',
};
