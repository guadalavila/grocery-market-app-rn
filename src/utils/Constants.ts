export const STORAGE_KEY = {
    FAVORITE: '@favorite',
};
