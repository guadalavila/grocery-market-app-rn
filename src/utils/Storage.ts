import AsyncStorage from '@react-native-async-storage/async-storage';

export const storeData = async (key: string, value: string) => {
    try {
        const jsonValue = JSON.stringify(value);
        await AsyncStorage.setItem(key, jsonValue);
    } catch (e) {}
};

export const getData = async (key: string) => {
    try {
        const data = await AsyncStorage.getItem(key);
        return data ? JSON.parse(data) : undefined;
    } catch (e) {
        // error reading value
    }
};
