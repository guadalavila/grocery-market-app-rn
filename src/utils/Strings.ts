export const Strings = {
    appName: 'Grocery Market',
    hello: 'Hola!',
    welcome: 'Bienvenido!',
    viewMore: 'Ver MÃ¡s',
    search: 'Buscar...',

    cart: {
        label: 'Carrito',
        empty: 'No tenes productos en el carrito',
        checkout: 'Terminar Compra',
        add: 'Agregar al carrito',
    },

    addedProduct: 'Producto Agregado!',

    fruits: {
        label: 'Frutas',
    },
    vegetables: {
        label: 'Verduras',
    },

    favorites: {
        label: 'Mis Favoritos',
        empty: 'No tenes productos como favoritos',
    },
};
