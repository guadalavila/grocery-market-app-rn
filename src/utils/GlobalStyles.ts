import { StyleSheet } from 'react-native';
import { Colors } from './Colors';

export const GlobalStyles = StyleSheet.create({
    container: {
        flex: 1,
    },
    shadowCard: {
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 4,
    },
    row: {
        flexDirection: 'row',
    },
    justBtw: {
        justifyContent: 'space-between',
    },
    main: {
        backgroundColor: Colors.background,
    },
});

export const flex = (x: number) => {
    return {
        flex: x,
    };
};

export const Sizes = {
    XXS: 2,
    XS: 4,
    S: 8,
    M: 16,
    L: 24,
    XL: 30,
    XXL: 40,
    XXXL: 55,
};

export const FontSize = {
    XS: 8,
    S: 10,
    M: 14,
    L: 18,
    XL: 22,
    XXL: 26,
    XXXL: 32,
};
