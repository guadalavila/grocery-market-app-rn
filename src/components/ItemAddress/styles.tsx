import React from 'react';
import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { FontSize } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    changeText: {
        color: Colors.primary,
        fontWeight: '700',
        textDecorationLine: 'underline',
    },
    buttonChange: {
        alignSelf: 'center',
    },
    addressText: {
        fontWeight: 'bold',
        color: Colors.black,
        fontSize: FontSize.L,
    },
    countryText: {
        color: Colors.black,
        fontSize: FontSize.M,
    },
});
