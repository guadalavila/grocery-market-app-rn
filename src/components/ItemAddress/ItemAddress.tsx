import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { Address } from '../../models/Address';
import { Colors } from '../../utils/Colors';
import { GlobalStyles } from '../../utils/GlobalStyles';
import Box from '../Box/Box';
import { styles } from './styles';

interface IItemAddressProps {
    address: Address;
    selected: boolean;
    onPress: () => void;
}

const ItemAddress = ({ address: { address, city, country, codePostal }, selected, onPress }: IItemAddressProps) => {
    return (
        <TouchableOpacity testID='button' style={GlobalStyles.containerWhite} activeOpacity={0.7} onPress={onPress}>
            <Box flexDirection='row' justifyContent='space-between'>
                <Box>
                    <Text style={styles.addressText}>{address} </Text>
                    <Text style={styles.countryText}>
                        {city}, {country}
                    </Text>
                    <Text>{codePostal}</Text>
                </Box>
                <Box alignSelf='center' background={Colors.grey200} borderRadius={6} width={30} height={30}>
                    {selected && (
                        <Icon
                            style={GlobalStyles.center}
                            testID='icon-check'
                            name='checkmark'
                            size={26}
                            color={Colors.primary}
                        />
                    )}
                </Box>
            </Box>
        </TouchableOpacity>
    );
};

export default ItemAddress;
