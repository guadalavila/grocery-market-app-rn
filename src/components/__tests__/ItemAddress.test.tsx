import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import ItemAddress from '../ItemAddress/ItemAddress';
import { Colors } from '../../utils/Colors';

describe('test ItemAddress component', () => {
    const addressMock = {
        id: '3xx68afc-c609-48d3-a4f8-fbd91aa97f67',
        address: 'Av Belgrano 1440',
        city: 'Salta',
        country: 'Argentina',
        codePostal: '4400',
        selected: true,
    };
    const onPressMock = jest.fn();

    test('should match with snapshot', () => {
        const component = render(<ItemAddress address={addressMock} selected onPress={onPressMock} />);
        expect(component).toMatchSnapshot();
    });

    test('should render component', () => {
        render(<ItemAddress address={addressMock} selected onPress={onPressMock} />);
        expect(screen).toBeTruthy();
    });
    test('should render texts', () => {
        render(<ItemAddress address={addressMock} selected onPress={onPressMock} />);
        expect(screen.getByText('Av Belgrano 1440')).toBeTruthy();
        expect(screen.getByText('Salta, Argentina')).toBeTruthy();
        expect(screen.getByText('4400')).toBeTruthy();
    });

    test('should render icon when select', () => {
        render(<ItemAddress address={addressMock} selected onPress={onPressMock} />);
        expect(screen.getByTestId('icon-check')).toBeTruthy();
        expect(screen.getByTestId('icon-check').props.name).toBe('checkmark');
        expect(screen.getByTestId('icon-check').props.size).toBe(26);
        expect(screen.getByTestId('icon-check').props.color).toBe(Colors.primary);
    });

    test('should call fn', () => {
        render(<ItemAddress address={addressMock} selected onPress={onPressMock} />);
        const btn = screen.getByTestId('button');
        fireEvent.press(btn);
        expect(onPressMock).toHaveBeenCalled();
    });
});
