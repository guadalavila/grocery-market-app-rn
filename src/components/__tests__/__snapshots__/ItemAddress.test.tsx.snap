// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ItemAddress component should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "marginBottom": 24,
      "marginHorizontal": 24,
      "marginTop": 8,
      "opacity": 1,
      "padding": 8,
      "paddingHorizontal": 16,
    }
  }
  testID="button"
>
  <View
    style={
      Object {
        "alignSelf": undefined,
        "backgroundColor": "transparent",
        "borderRadius": undefined,
        "display": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 0,
        "marginBottom": 0,
        "marginHorizontal": 0,
        "marginTop": 0,
        "marginVertical": 0,
        "padding": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "paddingTop": 0,
        "paddingVertical": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": undefined,
          "backgroundColor": "transparent",
          "borderRadius": undefined,
          "display": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "justifyContent": undefined,
          "margin": 0,
          "marginBottom": 0,
          "marginHorizontal": 0,
          "marginTop": 0,
          "marginVertical": 0,
          "padding": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "paddingTop": 0,
          "paddingVertical": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        Av Belgrano 1440
         
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 15,
          }
        }
      >
        Salta
        , 
        Argentina
      </Text>
      <Text>
        4400
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#E8E8E8",
          "borderRadius": 6,
          "display": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "justifyContent": undefined,
          "margin": 0,
          "marginBottom": 0,
          "marginHorizontal": 0,
          "marginTop": 0,
          "marginVertical": 0,
          "padding": 0,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "paddingTop": 0,
          "paddingVertical": 0,
        }
      }
    >
      <MockedIoniconsIcons
        color="#68B984"
        name="checkmark"
        size={26}
        testID="icon-check"
      />
    </View>
  </View>
</View>
`;
